core/cache/controller.go

```diff
@@ -138,6 +138,20 @@ func (c *Controller) Model(uuid string) (*Model, error) {
 	return model, nil
 }
 
+// Application returns the application for the input model UUID and app name.
+// If the model or application are not found, a NotFoundError is returned.
+func (c *Controller) Application(modelUUID, appName string) (*Application, error) {
+	model, err := c.Model(modelUUID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	c.mu.Lock()
```

Comment:
We don't need this Lock and Unlock right? Because the model locks to get the app?